### PR TITLE
kaidan: init at 0.8.0

### PR DESCRIPTION
**Pull request by astro**
###### Motivation for this change
New XMPP client

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125884` at 891dd558 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>kaidan</li>
    <li>qxmpp</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`lgpl21` is a deprecated license, please check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/development/libraries/qxmpp/default.nix:36:5:
```
   |
36 |     license = licenses.lgpl21;
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/development/libraries/qxmpp/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    gst-plugins-bad
+    gst-plugins-base
+    gst-plugins-good
+  ]);
+  cmakeFlags = [ "-DWITH_GSTREAMER=ON" ];
+
+  meta = with lib; {
+    description = "Cross-platform C++ XMPP client and server library";
+    homepage = "https://github.com/qxmpp-project/qxmpp";
+    license = licenses.lgpl21;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qxmpp/default.nix:36]: lgpl21Plus or lgpl21Only?

File: pkgs/development/libraries/qxmpp/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  nativeBuildInputs = [
+    cmake
+  ]
+  ++ lib.optionals withGstreamer (with gst_all_1; [
+    pkg-config
+    gstreamer
+    gst-plugins-bad
+    gst-plugins-base
+    gst-plugins-good
+  ]);
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qxmpp/default.nix:30]:
```suggestion
  ] ++ lib.optionals withGstreamer (with gst_all_1; [
    pkg-config
  ]);
  
  buildInputs = lib.optionals withGstreamer [
    gstreamer
    gst-plugins-bad
    gst-plugins-base
    gst-plugins-good
  ];
```

File: pkgs/development/libraries/qxmpp/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{ mkDerivation
+, lib
+, fetchFromGitHub
+, cmake
+, pkg-config
+, withGstreamer ? true
+, gst_all_1 ? null
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qxmpp/default.nix:7]:
```suggestion
, gst_all_1
```

File: pkgs/development/libraries/qxmpp/default.nix (hunk 1 above)
Comment (astro) [reply pkgs/development/libraries/qxmpp/default.nix:36]: Just 2.1: https://github.com/qxmpp-project/qxmpp/blob/master/LICENSE.LGPL

**Comment by astro** [comment]
Thank you for the review.

I need to admit that A/V call functionality is broken in this build:
```
file:///nix/store/k0159z503v1hkn7w97098rp37r1iz8qf-kirigami2-5.81.0/lib/qt-5.15.2/qml/org/kde/kirigami.2/Page.qml:355: Error: Cannot assign QObject* to PageRow_QMLTYPE_6*
file:///nix/store/k0159z503v1hkn7w97098rp37r1iz8qf-kirigami2-5.81.0/lib/qt-5.15.2/qml/org/kde/kirigami.2/Page.qml:355: Error: Cannot assign QObject* to PageRow_QMLTYPE_6*
Error: cannot create camera service, the 'camerabin' plugin is missing for GStreamer 1.18.
Please install the 'bad' GStreamer plugin package.
```

File: pkgs/development/libraries/qxmpp/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/qxmpp/default.nix:42]: Actually it is lgpl21Plus https://github.com/qxmpp-project/qxmpp/blob/master/src/client/QXmppArchiveManager.cpp#L15. The license texts for both are the same.

File: pkgs/development/libraries/qxmpp/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  ]
+  ++ lib.optionals withGstreamer (with gst_all_1; [
+    pkg-config
+    gstreamer
+    gst-plugins-bad
+    gst-plugins-base
+    gst-plugins-good
+  ]);
+  cmakeFlags = [
+    "-DWITH_GSTREAMER=ON"
```
Comment (polygon) [inline pkgs/development/libraries/qxmpp/default.nix:34]: This option should only be added when `withGstreamer` is `true`

**Comment by polygon** [comment]
##### Reviewed points

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [ ] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] phases are respected

File: pkgs/development/libraries/qxmpp/default.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    "-DBUILD_EXAMPLES=false"
+    "-DBUILD_TESTS=false"
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qxmpp/default.nix:37]:
```suggestion
  cmakeFlags = [
    "-DBUILD_EXAMPLES=false"
    "-DBUILD_TESTS=false"
  ] ++ lib.optionals withGstreamer [
    "-DWITH_GSTREAMER=ON"
  ];
```

**Comment by astro** [comment]
Thank you for reviewing.

**Pull request merged**